retrieval/series_cache.go

```diff
@@ -393,12 +393,15 @@ func (c *seriesCache) refresh(ctx context.Context, ref uint64) error {
 	if metadata == nil {
 		// The full name didn't turn anything up. Check again in case it's a summary or histogram without
 		// the metric name suffix.
-		var ok bool
-		if baseMetricName, suffix, ok = stripComplexMetricSuffix(metricName); ok {
+		var useBaseMetricName, ok bool
+		if baseMetricName, suffix, useBaseMetricName, ok = stripComplexMetricSuffix(metricName); ok {
```

Comment:
I don't think whether to use baseMetricName is up to `stripComplextMetricsSuffix`. histogram metric also use baseMetricName as well. 

I think we need to use baseMetricName if one of 1-a) and 1-b) is true AND 2) is true:

1-a) the type of series is histogram
1-b) the type of series is counter, and it has with "_total" suffix
2) the metadata is retrievable only with `baseMetricName`.

I think we should keep the existing interface and implementation of `stripComplextMetricSuffix`, and update https://github.com/Stackdriver/stackdriver-prometheus-sidecar/pull/146/files#diff-e9d6484b8661777dd3d9060c77d46c35R433 as:

```
	case textparse.MetricTypeCounter:
		ts.MetricKind = metric_pb.MetricDescriptor_CUMULATIVE
		ts.ValueType = metric_pb.MetricDescriptor_DOUBLE
		if baseMetricName != "" && suffix == metricSuffixTotal {
			ts.Metric.Type = c.getMetricType(c.metricsPrefix, baseMetricName)
		}
```

what do you think?

Reply:
SGTM, updated.